assets/js/blocks/collection-filters/utils.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { ProductCollectionQuery } from '@woocommerce/blocks/product-collection/types';
+
+function taxonomyMapper( taxonomy: string ) {
+	const map: Record< string, string > = {
+		product_tag: 'tag',
+		product_cat: 'cat',
+	};
+
+	return taxonomy in map ? map[ taxonomy ] : `_unstable_tax_${ taxonomy }`;
+}
+
+export function formatQuery( query: ProductCollectionQuery ) {
+	if ( ! query ) {
+		return {};
+	}
+
+	const sharedParams: Array< keyof ProductCollectionQuery > = [
+		'exclude',
+		'offset',
+		'order',
+		'search',
+	];
+
+	const mappedParams: { key: keyof ProductCollectionQuery; map: string }[] = [
+		{ key: 'orderBy', map: 'orderby' },
+		{ key: 'pages', map: 'page' },
+		{ key: 'parents', map: 'parent' },
+		{ key: 'perPage', map: 'per_page' },
+		{ key: 'woocommerceStockStatus', map: 'stock_status' },
+		{ key: 'woocommerceOnSale', map: 'on_sale' },
+		{ key: 'woocommerceHandPickedProducts', map: 'include' },
+	];
+
+	return Object.assign(
```

Comment:
Why `Object.assign` instead of just spreading over an empty object?

```js
{
   ...sharedParams.reduce(),
   ...mappedParams.reduce()
}
```

Reply:
Because we want to receive an object. If we don't use `Object.assign`, then we receive an object of object like this:
```
{
	{ per_page: 10 },
	{ orderby: 'title' },
}
```

Reply:
Mm, are you sure? It seems to me that if you use spread operator over an object that should be flat, no?

Reply:
> Mm, are you sure?

Yes, if you look at the returned value from the map functions, they're arrays of objects: 

```
		...sharedParams.map(
			( key ) => key in query && { [ key ]: query[ key ] }
		),
```

Sure, I can use `reduce` here, but I always find `map` is easier to read than `reduce`.

Reply:
Ah, yea, I meant to ask if you were sure about that being the result when using `reduce`. Ok, then, if it's your preference, you can use `map`. I personally try to be more explicit, and `reduce` helps me doing that. But that's not a hard recommendation, just an interesting conversation.